### PR TITLE
builder: use builtin Clang when building statically

### PR DESCRIPTION
**Pull request by aykevl**
This will be a huge help for people installing TinyGo that don't have LLVM/Clang 9 already installed and in the `$PATH` variable.

Replaces #184.

I added an extra note to the LICENSE part of the README, would that be enough to cover the new code under the different license? I don't think adding it to the LICENSE file would be necessary as the files themselves already say that they are under a different license.

### CODE REVIEW

**Comment by deadprogram** [comment]
I would suggest also adding the additional license info to https://github.com/tinygo-org/tinygo/blob/master/LICENSE

**Comment by deadprogram** [comment]
OMG it compiles!!! 

One question is do we still need `lld` linker install, since this replaces clang-9?

File: builder/compiler-builtin.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// +build byollvm
+
+package builder
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	//"path/filepath"
```
Comment (deadprogram) [inline builder/compiler-builtin.go]: Should remove this probably.

**Comment by deadprogram** [comment]
@aykevl please note my question and comment on this PR. Thank you.

File: builder/compiler-builtin.go (hunk 1 above)
Comment (aykevl) [reply builder/compiler-builtin.go]: Oops, fixed.

**Comment by aykevl** [comment]
> I would suggest also adding the additional license info to https://github.com/tinygo-org/tinygo/blob/master/LICENSE

What do you mean exactly? Copy the entire license text (something I'd rather not do) or just say something like this?

> TinyGo includes portions of LLVM, which is under the Apache License v2.0 with LLVM Exceptions.
> See https://llvm.org/LICENSE.txt for license information.

---

> One question is do we still need `lld` linker install, since this replaces clang-9?

We already bundle `lld` in static/release (`make`) builds for quite some time, so no. For `go install` builds nothing has changed.

**Comment by deadprogram** [comment]
Just adding your propsed excerpt to our license is sufficient to comply with the Apache 2.0 license.

**Comment by aykevl** [comment]
Updated.

**Comment by deadprogram** [comment]
Thanks for the update. Now merging.

**Comment by aykevl** [comment]
Great!

**Pull request merged**